src/lib.rs

```diff
@@ -113,12 +113,11 @@ where
         Ok(())
     }
 
-    /// Initialize the display controller
-    pub fn init<Delay: DelayNs>(
+    #[inline]
+    fn do_init<Delay: DelayNs>(
```

Comment:
It's quite a bit of boilerplate code. 
How about using drop to achieve the same result?
```
struct ChipSelect {
  ...
}

impl Drop for ChipSelect {
    fn drop(&mut self) {
        ...
    }
}
```

Reply:
There will be some trickery to pass all the references needed to the new struct, but should be doable.

Reply:
https://doc.rust-lang.org/rust-by-example/trait/drop.html

Reply:
I have experimented with that and I don't think it's easily doable, because both the operation itself and the Drop need a mutable reference to self.

But anyway, what do you mean by boilerplate code? It's just one function call and will be inlined. The code itself is exactly as before.